src/transaction/types/utxo_id.rs

```diff
@@ -0,0 +1,113 @@
+use std::str::FromStr;
+
+use fuel_types::Bytes32;
+
+use crate::TxId;
+
+/// Identification of unspend transaction output.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct UtxoId {
+    /// transaction id
+    pub tx_id: TxId,
```

Comment:
The crate don't use `pub` accessors because this leads to coupled code. Use `pub const fn x(&self) -> X` instead